common/FileOps.h

```diff
@@ -43,6 +43,10 @@ class FileOps final {
     // prevent the caller from needing to try/catch removeDir.
     static bool removeEmptyDir(const std::string &path);
 
+    // Removes directories recursively, but they have to be completely empty. Useful for cleaning up dangling
+    // autoloader shim directories after an autogen run.
+    static void removeEmptyDirsRecursively(const std::string &dirPath);
```

Comment:
This seems like an oddly specific piece of code to have in common/FileOps.h, especially given that it's used in one file.

Can we move this to a helper function in the file that needs it?

Reply:
Ok, but would it be reasonable if it's implemented in `common.cc`? I wanted it to be closer to some of the filesystem primitives that are being already used in that file...since it's using the same ones like `readdir`. Or do you want header and implementation both moved?

Reply:
eh you're right that it's probably better to keep it where it is.